cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHandler.java

```diff
@@ -287,7 +287,7 @@ public void delete(HttpRequest request, HttpResponder responder,
                      @PathParam("stream") String stream) throws Exception {
     Id.Stream streamId = Id.Stream.from(namespaceId, stream);
     checkStreamExists(streamId);
-
+    streamWriter.close(streamId);
```

Comment:
So this only for Standalone? In distributed mode, there can be more than one `ConcurrentStreamWriter` running (one per Yarn container) and only one of those containers will get the `delete` REST call.